common/djangoapps/student/models_api.py

```diff
@@ -62,3 +67,76 @@ def get_phone_number(user_id):
         log.exception(exception)
         return None
     return student.phone_number or None
+
+
+def is_username_retired(username):
+    """
+    Checks to see if the given username has been previously retired
+    """
+    locally_hashed_usernames = user_util.get_all_retired_usernames(
+        username,
+        settings.RETIRED_USER_SALTS,
+        settings.RETIRED_USERNAME_FMT
+    )
+
+    # TODO: Revert to this after username capitalization issues detailed in
+    # PLAT-2276, PLAT-2277, PLAT-2278 are sorted out:
+    # return User.objects.filter(username__in=list(locally_hashed_usernames)).exists()
+
+    # Avoid circular import issues
+    from openedx.core.djangoapps.user_api.models import UserRetirementStatus
+
+    # Sandbox clean builds attempt to create users during migrations, before the database
+    # is stable so UserRetirementStatus may not exist yet. This workaround can also go
+    # when we are done with the username updates.
+    try:
+        return _User.objects.filter(username__in=list(locally_hashed_usernames)).exists() or \
+            UserRetirementStatus.objects.filter(original_username=username).exists()
```

Comment:
You cannot reference this here. You are in django app in the common area. The UserRequirementStatus is in the openedx/core/djangoapp area. The openedx/core/djangoapp depend on the student app here. Referencing back is a circular reference.